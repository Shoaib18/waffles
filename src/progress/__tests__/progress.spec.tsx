import { render } from '@testing-library/react';

import { Progress } from '../index';

describe('Progress', () => {
  it('renders a progress bar', () => {
    const { getByTestId } = render(<Progress value="40" max="100" />);
    const progress = getByTestId('progress-wrapper');

    expect(progress).toBeInTheDocument();
    expect(progress).toMatchSnapshot();
  });

  it('renders snapshot of an inverted progress bar', () => {
    const { getByTestId } = render(<Progress inverted value="40" max="100" />);
    const progress = getByTestId('progress-wrapper');

    expect(progress).toBeInTheDocument();
    expect(progress).toMatchSnapshot();
  });

  it('renders snapshot of a progress bar with steps', () => {
    const { getByTestId } = render(<Progress showSteps value="2" max="5" />);
    const progress = getByTestId('progress-wrapper');

    expect(progress).toBeInTheDocument();
    expect(progress).toMatchSnapshot();
  });
});
