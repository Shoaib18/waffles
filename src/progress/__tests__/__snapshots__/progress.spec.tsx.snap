// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders snapshot of a progress 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.emotion-1 {
  width: 100%;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-2 {
  background-color: rgba(5, 25, 45, 0.15);
  border: none;
  border-radius: 4px;
  -webkit-clip-path: url(#progress-uhTXNP-clip);
  clip-path: url(#progress-uhTXNP-clip);
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: 8px;
  width: 100%;
}

.emotion-2::-webkit-progress-bar {
  background-color: inherit;
  border-radius: 4px;
}

.emotion-2::-moz-progress-bar {
  background-color: #03EF62;
  border-radius: 4px;
  -webkit-transition: width 1s ease-in-out;
  transition: width 1s ease-in-out;
}

.emotion-2::-webkit-progress-value {
  background-color: #03EF62;
  border-radius: 4px;
  -webkit-transition: width 1s ease-in-out;
  transition: width 1s ease-in-out;
}

.emotion-3 {
  color: #65707C;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 12px;
  line-height: 1;
}

<div
  class="emotion-0"
  data-testid="progress-wrapper"
>
  <div
    class="emotion-1"
  >
    <progress
      class="emotion-2"
      id="progress-uhTXNP"
      max="100"
      value="40"
    />
  </div>
  <label
    class="emotion-3"
    for="progress-uhTXNP"
  >
    40%
  </label>
</div>
`;

exports[`Progress renders snapshot of a progress with steps 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.emotion-1 {
  width: 100%;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-2 {
  position: absolute;
  width: 100%;
  height: 1px;
}

.emotion-3 {
  x: calc(0);
  width: calc(20%);
  height: 8px;
}

.emotion-4 {
  x: calc(20% + 4px);
  width: calc(20% - 4px);
  height: 8px;
}

.emotion-5 {
  x: calc(40% + 4px);
  width: calc(20% - 4px);
  height: 8px;
}

.emotion-6 {
  x: calc(60% + 4px);
  width: calc(20% - 4px);
  height: 8px;
}

.emotion-7 {
  x: calc(80% + 4px);
  width: calc(20% - 4px);
  height: 8px;
}

.emotion-8 {
  background-color: rgba(5, 25, 45, 0.15);
  border: none;
  border-radius: 4px;
  -webkit-clip-path: url(#progress-IIzHra-clip);
  clip-path: url(#progress-IIzHra-clip);
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: 8px;
  width: 100%;
}

.emotion-8::-webkit-progress-bar {
  background-color: inherit;
  border-radius: 4px;
}

.emotion-8::-moz-progress-bar {
  background-color: #03EF62;
  border-radius: 4px;
  -webkit-transition: width 1s ease-in-out;
  transition: width 1s ease-in-out;
}

.emotion-8::-webkit-progress-value {
  background-color: #03EF62;
  border-radius: 4px;
  -webkit-transition: width 1s ease-in-out;
  transition: width 1s ease-in-out;
}

.emotion-9 {
  color: #65707C;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 12px;
  line-height: 1;
}

<div
  class="emotion-0"
  data-testid="progress-wrapper"
>
  <div
    class="emotion-1"
  >
    <svg
      class="emotion-2"
    >
      <defs>
        <clippath
          id="progress-IIzHra-clip"
        >
          <rect
            class="emotion-3"
            rx="4"
          />
          <rect
            class="emotion-4"
            rx="4"
          />
          <rect
            class="emotion-5"
            rx="4"
          />
          <rect
            class="emotion-6"
            rx="4"
          />
          <rect
            class="emotion-7"
            rx="4"
          />
        </clippath>
      </defs>
    </svg>
    <progress
      class="emotion-8"
      id="progress-IIzHra"
      max="5"
      value="2"
    />
  </div>
  <label
    class="emotion-9"
    for="progress-IIzHra"
  >
    2/5
  </label>
</div>
`;

exports[`Progress renders snapshot of an inverted progress 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.emotion-1 {
  width: 100%;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-2 {
  background-color: rgba(255, 255, 255, 0.2);
  border: none;
  border-radius: 4px;
  -webkit-clip-path: url(#progress-870DHg-clip);
  clip-path: url(#progress-870DHg-clip);
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: 8px;
  width: 100%;
}

.emotion-2::-webkit-progress-bar {
  background-color: inherit;
  border-radius: 4px;
}

.emotion-2::-moz-progress-bar {
  background-color: #03EF62;
  border-radius: 4px;
  -webkit-transition: width 1s ease-in-out;
  transition: width 1s ease-in-out;
}

.emotion-2::-webkit-progress-value {
  background-color: #03EF62;
  border-radius: 4px;
  -webkit-transition: width 1s ease-in-out;
  transition: width 1s ease-in-out;
}

.emotion-3 {
  color: #9BA3AB;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 12px;
  line-height: 1;
}

<div
  class="emotion-0"
  data-testid="progress-wrapper"
>
  <div
    class="emotion-1"
  >
    <progress
      class="emotion-2"
      id="progress-870DHg"
      max="100"
      value="40"
    />
  </div>
  <label
    class="emotion-3"
    for="progress-870DHg"
  >
    40%
  </label>
</div>
`;
